app/uk/gov/hmrc/tariffclassificationfrontend/controllers/CaseController.scala

```diff
@@ -73,7 +74,7 @@ class CaseController @Inject()(authenticatedAction: AuthenticatedAction,
     )
   }
 
-  def rulingDetails(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
+  def rulingDetails(reference: String): Action[AnyContent] = verify.caseExistsAndCheckPermissions(reference).async { implicit request =>
     getCaseAndRenderView(
```

Comment:
If we have the case on the request we dont need the getCase call any more in the controller